### ISSUE TITLE
Patch allowed on targaryen, not on firebase

### ISSUE THREAD

**Issue by SamyPesse** [goldibex/targaryen#134]
Targaryen is allowing patch that should be rejected.

The following patch is invalid because a key `-KwzSBcjqM_ydHz6415J` is forgotten after the `invites/`:

```js
{ 'orgs/org1/invites/':
       { uid: '-KwzSBcjqM_ydHz6415J',
         role: 'admin',
         createdAt: 1508598138982 } }
```

The Firebase rules are:

```
        ....
        "invites": {
          "$key2": {
            ".validate": "newData.hasChildren(['uid', 'role', 'createdAt']) && $key2 == newData.child('uid').val()",
            "uid": {
              ".validate": "newData.isString()"
            },
            "role": {
              ".validate": "newData.isString() && (newData.val() == 'reader' || newData.val() == 'writer' || newData.val() == 'reviewer' || newData.val() == 'admin')"
            },
            "createdAt": {
              ".validate": "newData.isNumber()"
            },
            "$other": {
              ".validate": "false"
            },
            ".read": "true"
          },
          ".validate": "newData.hasChildren() && newData.parent().parent().parent().child('orgs').child($orgID).child('infos').val() != null",
          ".read": "root.child('orgs').child($orgID).child('members').child(auth.uid).child('role').val() == 'admin'",
          ".write": "newData.parent().parent().parent().child('orgs').child($orgID).child('members').child(auth.uid).child('role').val() == 'admin'"
        }
       ....
```

The update is accepted on Targaryen (log of `data.update('/', update)`:

```
Result {
      path: '/',
      auth: { uid: 'user3' },
      type: 'patch',
      permitted: true,
      validated: true,
      database:
       Database { ... },
      newDatabase:
       Database { ... },
      newValue:
       { 'orgs/org1/invites/':
          { uid: '-KwzSBcjqM_ydHz6415J',
            role: 'admin',
            createdAt: 1508598138982 } } }
```

But on Firebase, the patch is rejected (as it should):

<img width="1215" alt="screen shot 2017-10-21 at 17 05 55" src="https://user-images.githubusercontent.com/845425/31853024-2bd5427c-b682-11e7-897c-0b9d089f5b55.png">

**Comment by dinoboff** [comment]
I can reproduce it, and the problem seems to be with the trailing slash.

**Comment by dinoboff** [comment]
@SamyPesse Should be fixed in [v3.0.4](https://github.com/goldibex/targaryen/releases/tag/v3.0.4).
